{"variants":[{"traits":[{"interfaceLanguage":"swift"}],"paths":["\/documentation\/swifterrorkit\/traceerrorattribute\/errorattribute-implementations"]}],"identifier":{"url":"doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit\/TraceErrorAttribute\/ErrorAttribute-Implementations","interfaceLanguage":"swift"},"schemaVersion":{"major":0,"patch":0,"minor":3},"topicSections":[{"identifiers":["doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit\/TraceErrorAttribute\/debugInfo(_:)","doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit\/TraceErrorAttribute\/tag(_:)","doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit\/TraceErrorAttribute\/trace(_:_:_:)"],"title":"Type Methods","generated":true}],"hierarchy":{"paths":[["doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit","doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit\/TraceErrorAttribute"]]},"metadata":{"modules":[{"name":"SwiftErrorKit"}],"role":"collectionGroup","title":"ErrorAttribute Implementations"},"kind":"article","sections":[],"references":{"doc://SwiftErrorKit/documentation/SwiftErrorKit":{"title":"SwiftErrorKit","abstract":[{"text":"Error attribution and commons.","type":"text"}],"kind":"symbol","url":"\/documentation\/swifterrorkit","type":"topic","role":"collection","identifier":"doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit"},"doc://SwiftErrorKit/documentation/SwiftErrorKit/TraceErrorAttribute/trace(_:_:_:)":{"identifier":"doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit\/TraceErrorAttribute\/trace(_:_:_:)","url":"\/documentation\/swifterrorkit\/traceerrorattribute\/trace(_:_:_:)","conformance":{"conformancePrefix":[{"text":"Conforms when","type":"text"}],"availabilityPrefix":[{"type":"text","text":"Available when"}],"constraints":[{"type":"codeVoice","code":"Self"},{"type":"text","text":" is "},{"code":"TraceErrorAttribute","type":"codeVoice"},{"text":".","type":"text"}]},"type":"topic","fragments":[{"text":"static","kind":"keyword"},{"text":" ","kind":"text"},{"kind":"keyword","text":"func"},{"kind":"text","text":" "},{"kind":"identifier","text":"trace"},{"kind":"text","text":"("},{"kind":"typeIdentifier","preciseIdentifier":"s:s12StaticStringV","text":"StaticString"},{"kind":"text","text":", "},{"preciseIdentifier":"s:Su","text":"UInt","kind":"typeIdentifier"},{"text":", ","kind":"text"},{"text":"UInt","preciseIdentifier":"s:Su","kind":"typeIdentifier"},{"text":") -> ","kind":"text"},{"kind":"typeIdentifier","text":"Self"}],"abstract":[],"title":"trace(_:_:_:)","role":"symbol","kind":"symbol"},"doc://SwiftErrorKit/documentation/SwiftErrorKit/TraceErrorAttribute/debugInfo(_:)":{"identifier":"doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit\/TraceErrorAttribute\/debugInfo(_:)","url":"\/documentation\/swifterrorkit\/traceerrorattribute\/debuginfo(_:)","type":"topic","fragments":[{"kind":"keyword","text":"static"},{"text":" ","kind":"text"},{"text":"func","kind":"keyword"},{"text":" ","kind":"text"},{"text":"debugInfo","kind":"identifier"},{"kind":"text","text":"<each "},{"kind":"genericParameter","text":"T"},{"kind":"text","text":">(repeat each "},{"kind":"typeIdentifier","text":"T"},{"kind":"text","text":") -> "},{"kind":"typeIdentifier","text":"DebugInfoErrorAttribute","preciseIdentifier":"s:13SwiftErrorKit09DebugInfoB9AttributeV"},{"text":"<repeat each ","kind":"text"},{"text":"T","kind":"typeIdentifier"},{"kind":"text","text":">"}],"abstract":[],"title":"debugInfo(_:)","role":"symbol","kind":"symbol"},"doc://SwiftErrorKit/documentation/SwiftErrorKit/TraceErrorAttribute":{"identifier":"doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit\/TraceErrorAttribute","url":"\/documentation\/swifterrorkit\/traceerrorattribute","navigatorTitle":[{"text":"TraceErrorAttribute","kind":"identifier"}],"type":"topic","fragments":[{"text":"struct","kind":"keyword"},{"text":" ","kind":"text"},{"text":"TraceErrorAttribute","kind":"identifier"}],"abstract":[{"text":"An error attribute to track a source location.","type":"text"}],"title":"TraceErrorAttribute","role":"symbol","kind":"symbol"},"doc://SwiftErrorKit/documentation/SwiftErrorKit/TraceErrorAttribute/tag(_:)":{"identifier":"doc:\/\/SwiftErrorKit\/documentation\/SwiftErrorKit\/TraceErrorAttribute\/tag(_:)","role":"symbol","abstract":[],"url":"\/documentation\/swifterrorkit\/traceerrorattribute\/tag(_:)","fragments":[{"text":"static","kind":"keyword"},{"text":" ","kind":"text"},{"text":"func","kind":"keyword"},{"text":" ","kind":"text"},{"kind":"identifier","text":"tag"},{"kind":"text","text":"<"},{"text":"Tag","kind":"genericParameter"},{"kind":"text","text":">("},{"text":"Tag","kind":"typeIdentifier"},{"text":") -> ","kind":"text"},{"kind":"typeIdentifier","text":"Self"}],"type":"topic","title":"tag(_:)","kind":"symbol"}}}